packages/manager/apps/telecom/src/app/telecom/telephony/svaWallet/kyc/identity/overview-identity/overview-identity-edit/overview-identity-edit.controller.js

```diff
@@ -0,0 +1,180 @@
+        name: this.$translate.instant(
+          `telephony_billingAccount_svaWallet_kyc_country_${this.svaWallet.representative.countryOfBirth}`,
+        ),
+        value: this.svaWallet.representative.countryOfBirth,
+      },
+      countryOfResidence: {
+        name: this.$translate.instant(
+          `telephony_billingAccount_svaWallet_kyc_country_${this.svaWallet.representative.countryOfResidence}`,
+        ),
+        value: this.svaWallet.representative.countryOfResidence,
+      },
+    };
+
+    this.representativeCountrySelects = [
+      'countryOfBirth',
+      'countryOfResidence',
+      'nationality',
+    ];
+
+    this.streetNumberExtraEnum = [
+      '',
+      ...DIRECTORY_WAY_NUMBER_EXTRA_ENUM.map((val) => val.toUpperCase()),
+    ];
+
+    this.VALIDATORS = VALIDATORS;
+    this.requirements = false;
+
+    this.wallet = {
+      company: this.svaWallet.company,
+      representative: { ...this.svaWallet.representative, isBeneficiary: true },
+    };
+
+    this.loading = {
+      submit: false,
+    };
+
+    this.companyKindsValues = buildEnumList(
+      this.companyKinds,
+      'telephony_billingAccount_svaWallet_kyc_company_kind_',
+      this.$translate,
+    );
+
+    this.countriesValues = buildEnumList(
+      this.countryEnum.filter((countryCode) => countryCode !== 'UNKNOWN'),
+      'telephony_billingAccount_svaWallet_kyc_country_',
+      this.$translate,
+    );
+
+    this.onCompanyKindChange({ value: this.svaWallet.company.kind });
+  }
+
+  onCompanyKindChange({ value }) {
+    this.displayBeneficiaries = !DISALLOW_BENEFICIARIES_KINDS.includes(value);
+    this.forceRepresentativeIsBeneficiary = FORCE_REPRESENTAIVE_IS_BENEFICIARY_KINDS.includes(
+      value,
+    );
+    if (this.forceRepresentativeIsBeneficiary) {
+      this.wallet.representative.isBeneficiary = true;
+    }
+  }
+
+  formatWallet() {
+    delete this.wallet.representative.id;
+    const companyKind = getEnumItemValue(this.companyModel.kind);
+    const allowBeneficiaries = !DISALLOW_BENEFICIARIES_KINDS.includes(
+      companyKind,
+    );
+
+    const company = {
+      ...this.wallet.company,
+      kind: companyKind,
+      beneficiaries: allowBeneficiaries
+        ? this.wallet.company.beneficiaries
+        : [],
+    };
+
+    const representative = {
+      ...this.wallet.representative,
+      // convert selects items to values
+      ...this.representativeCountrySelects.reduce(
+        (values, field) => ({
+          ...values,
+          [field]: getEnumItemValue(this.representativeModel[field]),
+        }),
+        {},
+      ),
+      addressOfResidence: {
+        ...this.wallet.representative.addressOfResidence,
+        streetNumberExtra: this.representativeModel.streetNumberExtra?.toLowerCase(),
+      },
+    };
+
+    return {
+      ...this.wallet,
+      company,
+      representative,
+    };
+  }
+
+  submit() {
+    if (this.form.$invalid) {
+      return this.$q.reject();
+    }
```

Comment:
You should block submit action here https://github.com/ovh/manager/pull/6509/files?diff=split&w=1#diff-0f4f780830bb99be655edf6cf94191c9ac9792b7ea0555f243f0aa5c64c8e244R372, to avoid user clicking and not making action intended